### PR TITLE
#126 customScript attribute in button component

### PR DESCRIPTION
**Pull request by krixerx**
#126 customScript attribute in button component

### CODE REVIEW

**Comment by travist** [comment]
I am curious why this is a necessary feature when I believe you can achieve everything you want with using the events like this.

```
Formio.createForm('https://myproject.form.io/myform').then((form) => {
  form.on('customEvent', () => {
    // Execute custom code here....
  });
});
```

**Comment by krixerx** [comment]
The reason is to build complex component.
Edit view + grid. In every grid row there are buttons (edit, delete) and below grid "add new".
It is more convenient to add (edit, delete) javascript code directly into button instead of playing with events.
And event doesn't support parameters. It is difficult to say to event which datagrid row i want to edit on edit panel.

File: src/components/button/Button.js [hunk 1]
```diff
@@ -42,6 +42,9 @@ export class ButtonComponent extends BaseComponent {
       this.label = this.text(this.component.label);
       this.element.appendChild(this.label);
     }
+    if (this.component.customScript) {
+        this.customScript = this.text(this.component.customScript);
```
Comment (travist) [inline src/components/button/Button.js]: Why is this needed?  I don't see that this does anything other than create a DOM element that has the contents of the script. I believe this would still work without this script.

File: src/components/button/Button.js [hunk 2]
```diff
@@ -74,6 +77,9 @@ export class ButtonComponent extends BaseComponent {
         case 'reset':
           this.emit('resetForm');
           break;
+        case 'script':
+          eval(this.component.customScript);
```
Comment (travist) [inline src/components/button/Button.js]: Needs to be wrapped within a ```try { } catch { }``` statement.

File: src/components/complexdatagrid/ComplexDataGrid.js [hunk 3]
```diff
@@ -0,0 +1,368 @@
+import _each from 'lodash/each';
```
Comment (travist) [inline src/components/complexdatagrid/ComplexDataGrid.js]: This needs to be a separate pull request.
